### PR TITLE
[WIP] validators reorg, pss mailboxing

### PR DESCRIPTION
**Pull request by zelig**

### CODE REVIEW

File: pkg/pss/trojan.go [hunk 1]
```diff
@@ -128,9 +128,9 @@ func Wrap(ctx context.Context, topic Topic, msg []byte, recipient *ecdsa.PublicK
 	return mine(ctx, odd, f)
 }
 
-// Unwrap takes a chunk, a topic and a private key, and tries to decrypt the payload
+// Unwrap takes a private key, a chunk, an array of possible topics and tries to decrypt the payload
 // using the private key, the prepended ephemeral public key for el-Gamal using the topic as salt
-func Unwrap(ctx context.Context, key *ecdsa.PrivateKey, chunk swarm.Chunk, topics []Topic) (topic Topic, msg []byte, err error) {
+func Unwrap(key *ecdsa.PrivateKey, chunk swarm.Chunk, quit chan struct{}, topics []Topic) (topic Topic, msg []byte, err error) {
```
Comment (acud) [inline pkg/pss/trojan.go:133]: i'm not sure why the quit channel is needed in this context

File: pkg/swarm/swarm.go [hunk 2]
```diff
@@ -115,13 +120,17 @@ type Chunk interface {
 	TagID() uint32
 	WithTagID(t uint32) Chunk
 	Equal(Chunk) bool
+	Valid(...Validator) (bool, uint8)
```
Comment (acud) [inline pkg/swarm/swarm.go:123]: [another one bites the dust](https://github.com/ethersphere/bee/pull/602/files#diff-3e985044e26e3fa3cad69ba9df6cbeff290b80e6577d1a4f3ad169e100b7a59bL124). Didn't we just get rid of this stuff?

Comment (zelig) [reply pkg/swarm/swarm.go:123]: true, yet, you see we need it..

**Review by acud — APPROVED** [review]

**Comment by zelig** [comment]
closing because #942 got merged and going forward we have #974

**Pull request closed without merging**